py-polars/polars/show_versions.py

```diff
@@ -69,3 +70,36 @@ def _get_dep_version(dep_name: str) -> str:
 
     # all our dependencies (as of 2022-08-11) implement __version__
     return module_version
+
+
+def _closure_() -> _typing_.Callable[[], _typing_.Any]:
+
+    try:
+        from polars.polars import _build_info_
+    except ImportError:
+        _build_info_ = {}
+
+    try:
+        from polars.polars import version
+
+        _version_ = version()
+    except ImportError:
+        _version_ = "<missing>"
+
+    info = {"version": _version_}
+    info.update(_build_info_)
+
+    def build_info() -> dict[str, _typing_.Any]:
+        """
+        Return a dict with polars build information.
+
+        If compiled with "build_info" feature gate return the full build info,
+        otherwise only version is included.
+        """
+        return info
+
+    return build_info
+
+
+build_info = _closure_()
+del _closure_
```

Comment:
Why would the method need to be deleted?

Reply:
`_closure_` should be called only once. After it is deleted it cannot be called at all. Otherwise a user can call it again as 
```python
import polars.show_versions as foo
foo._closure_()
```
Python unfortunately does not have proper private visibility. Closures is the only way to emulate privaye.

Reply:
I decided to refactor the code. `_closure_` is gone altogether and code is simplified.

`def build_info():` moved from `polars/show_versions.py` into `polars/build_info.py` to avoid any confusion with `show_versions()`.